sibilant/specials.py

```diff
@@ -1325,22 +1325,53 @@ def special_yield_from(code, source, tc=False):
         msg = "Too few arguments to %s, %r" % (called_by, source)
         raise code.error(msg, source)
 
-    if not is_nil(rest):
-        msg = "Too many arguments to %s, %r" % (called_by, source)
-        raise code.error(msg, source)
-
     # the expression being yielded from is always NOT tco valid
     code.add_expression(value, False)
 
     # according to the dis page, I should only need to call
     # YIELD_FROM, but that always fails. If I look at a disassembly of
     # a simple yield from call in Python, I see that there's a const
     # None at TOS, with the generator below it whenever YIELD_FROM
-    # happens. I'm emulating that here.
+    # happens.
+
+    if not is_nil(rest):
+
+        try:
+            kw, arg = rest.unpack()
+
+        except ValueError:
+            # helpful comment describing helpful error message
+            msg = "Wrong arguments to %s, %r" % (called_by, source)
+            raise code.error(msg, source)
+
+        if kw == keyword("send"):
+            return arg
+
+        else:
+            msg = "Too many arguments to %s, %r" % (called_by, source)
+            raise code.error(msg, source)
+
+    else:
+        return None
+
+
+@special(_symbol_await)
+def special_await(code, source, tc=False):
+    send_value = _helper_yield_from(code, source)
+
+    code.pseudop_get_awaitable()
+    code.pseudop_const(send_value)
+    code.pseudop_yield_from()
+
+    return None
+
+
+@special(_symbol_yield_from)
+def special_yield_from(code, source, tc=False):
+    send_value = _helper_yield_from(code, source)
 
-    # it looks like the None is being sent to the iter!
     code.pseudop_get_yield_from_iter()
-    code.pseudop_const(None)
+    code.pseudop_const(send_value)
```

Comment:
this might need to be code.add_expression, because it might be ... an expression! Will work fine for None cases, but other things can't fit into consts

Reply:
Passing only immediate values was _definitely_ intentional. Definitely.